### ISSUE TITLE
map error

### ISSUE THREAD

**Issue by samth** [pycket/pycket#266]
Exits pycket unexpectedly
```
[samth@huor:~/.../racket-test-core/tests/racket (master) plt] pk 
Welcome to Pycket v7.1.0.6
> (map)
ERROR:
No matching arity in case-lambda
  while executing: (let ([repl-loop_1 (lambda () (let ([AppRand30905 (default-continuation-prompt-tag)]) (call-with-continuation-prompt (lambda () (let ([AppRator341 (current-prompt-read)]) (let ([v_2 (AppRator341)]) (if (eof-object? v_2) (void) (begin (call-with-values (lambda () (call-with-continuation-prompt (lambda () (let ([w_3 (cons '#%top-interaction v_2)][AppRator342 (LinkletVar(current-eval:NO-VAL))][if5491 (LinkletVar(1/syntax?:NO-VAL) v_2)]) (let ([AppRator342 AppRator342][AppRand30902 (if if5491 (let ([AppRand30901 (LinkletVar(1/datum->syntax:NO-VAL) #f w_3 v_2)]) (LinkletVar(namespace-syntax-introduce:NO-VAL) AppRand30901)) w_3)]) (AppRator342 AppRand30902)))))) (lambda results_4 [(let ([AppRand30903 (current-print)]) (for-each AppRand30903 results_4))])) (let ([AppRand30904 (default-continuation-prompt-tag)]) (abort-current-continuation AppRand30904))))))) AppRand30905 (lambda args_5 [(repl-loop_1)]))))]) (repl-loop_1))
  in function: (lambda () (let ([repl-loop_1 (lambda () (let ([AppRand30905 (default-continuation-prompt-tag)]) (call-with-continuation-prompt (lambda () (let ([AppRator341 (current-prompt-read)]) (let ([v_2 (AppRator341)]) (if (eof-object? v_2) (void) (begin (call-with-values (lambda () (call-with-continuation-prompt (lambda () (let ([w_3 (cons '#%top-interaction v_2)][AppRator342 (LinkletVar(current-eval:NO-VAL))][if5491 (LinkletVar(1/syntax?:NO-VAL) v_2)]) (let ([AppRator342 AppRator342][AppRand30902 (if if5491 (let ([AppRand30901 (LinkletVar(1/datum->syntax:NO-VAL) #f w_3 v_2)]) (LinkletVar(namespace-syntax-introduce:NO-VAL) AppRand30901)) w_3)]) (AppRator342 AppRand30902)))))) (lambda results_4 [(let ([AppRand30903 (current-print)]) (for-each AppRand30903 results_4))])) (let ([AppRand30904 (default-continuation-prompt-tag)]) (abort-current-continuation AppRand30904))))))) AppRand30905 (lambda args_5 [(repl-loop_1)]))))]) (repl-loop_1)))
RPython traceback:
  File "pycket_entry_point.c", line 47, in entry_point
  File "pycket_entry_point.c", line 1204, in actual_entry
  File "pycket_racket_entry.c", line 1667, in racket_entry
  File "pycket_base.c", line 545, in W_Object_call_interpret
  File "pycket_interpreter.c", line 22980, in interpret_one
  File "rpython_jit_metainterp.c", line 14972, in handle_jitexception_6
  File "pycket_interpreter_1.c", line 4736, in portal_6
  File "pycket_values.c", line 8485, in W_Closure_call_with_extra_info
  File "pycket_values.c", line 14804, in W_Closure__find_lam
Fatal RPython error: SchemeException
Aborted (core dumped)
```

**Comment by samth** [comment]
```
(map (lambda (x) (values 1 2)) (quote (1 2)))  =e=> ERROR:
wrong number of values
  while executing: (let ([if33540 (path? lam.val.18505)]) (if if33540 #f (let ([if33538 (string? lam.val.18505)]) (let ([if33539 (if if33538 (let ([if33537 (relative-path? lam.val.18505)]) (if if33537 #t (absolute-path? lam.val.18505))) #f)]) (if if33539 #f (raise-argument-error 'load path-string? lam.val.18505))))) (let ([if33541 (string? lam.val.18505)]) (let ([letone18506 (if if33541 (string->path lam.val.18505) lam.val.18505)]) (let ([let-void-0.18507 'uninitialized][let-void-1.18508 'uninitialized][let-void-2.18509 'uninitialized][(inst-val0.18510 inst-val1.18511 inst-val2.18512) (split-path letone18506)]) (set! CellRef(let-void-0.18507) inst-val0.18510) (set! CellRef(let-void-1.18508) inst-val1.18511) (set! CellRef(let-void-2.18509) inst-val2.18512) (with-continuation-mark parameterization-key (let ([AppRand5438 (continuation-mark-set-first #f parameterization-key)][if33542 (eq? CellRef(let-void-0.18507) 'relative)]) (let ([AppRand5438 AppRand5438][AppRand5439 (if if33542 (current-directory) (path->complete-path CellRef(let-void-0.18507)))]) (extend-parameterization AppRand5438 current-load-relative-directory AppRand5439))) (let ([AppRator172 (LinkletVar(current-load:#<parameter-procedure>))]) (AppRator172 letone18506 #f)))))))
  in function: (lambda (lam.val.18505) (let ([if33540 (path? lam.val.18505)]) (if if33540 #f (let ([if33538 (string? lam.val.18505)]) (let ([if33539 (if if33538 (let ([if33537 (relative-path? lam.val.18505)]) (if if33537 #t (absolute-path? lam.val.18505))) #f)]) (if if33539 #f (raise-argument-error 'load path-string? lam.val.18505))))) (let ([if33541 (string? lam.val.18505)]) (let ([letone18506 (if if33541 (string->path lam.val.18505) lam.val.18505)]) (let ([let-void-0.18507 'uninitialized][let-void-1.18508 'uninitialized][let-void-2.18509 'uninitialized][(inst-val0.18510 inst-val1.18511 inst-val2.18512) (split-path letone18506)]) (set! CellRef(let-void-0.18507) inst-val0.18510) (set! CellRef(let-void-1.18508) inst-val1.18511) (set! CellRef(let-void-2.18509) inst-val2.18512) (with-continuation-mark parameterization-key (let ([AppRand5438 (continuation-mark-set-first #f parameterization-key)][if33542 (eq? CellRef(let-void-0.18507) 'relative)]) (let ([AppRand5438 AppRand5438][AppRand5439 (if if33542 (current-directory) (path->complete-path CellRef(let-void-0.18507)))]) (extend-parameterization AppRand5438 current-load-relative-directory AppRand5439))) (let ([AppRator172 (LinkletVar(current-load:#<parameter-procedure>))]) (AppRator172 letone18506 #f))))))))
RPython traceback:
  File "pycket_entry_point.c", line 47, in entry_point
  File "pycket_entry_point.c", line 1204, in actual_entry
  File "pycket_racket_entry.c", line 2818, in racket_entry
  File "pycket_base.c", line 545, in W_Object_call_interpret
  File "pycket_interpreter.c", line 22980, in interpret_one
  File "rpython_jit_metainterp.c", line 14972, in handle_jitexception_6
  File "pycket_interpreter_1.c", line 4736, in portal_6
  File "pycket_interpreter.c", line 30858, in LetCont_plug_reduce
Fatal RPython error: SchemeException
Aborted (core dumped)
```

**Comment by samth** [comment]
```
(ormap (lambda (x) (values 1 2)) (quote (1 2)))  =e=> ERROR:
wrong number of values
  while executing: (let ([if33540 (path? lam.val.18505)]) (if if33540 #f (let ([if33538 (string? lam.val.18505)]) (let ([if33539 (if if33538 (let ([if33537 (relative-path? lam.val.18505)]) (if if33537 #t (absolute-path? lam.val.18505))) #f)]) (if if33539 #f (raise-argument-error 'load path-string? lam.val.18505))))) (let ([if33541 (string? lam.val.18505)]) (let ([letone18506 (if if33541 (string->path lam.val.18505) lam.val.18505)]) (let ([let-void-0.18507 'uninitialized][let-void-1.18508 'uninitialized][let-void-2.18509 'uninitialized][(inst-val0.18510 inst-val1.18511 inst-val2.18512) (split-path letone18506)]) (set! CellRef(let-void-0.18507) inst-val0.18510) (set! CellRef(let-void-1.18508) inst-val1.18511) (set! CellRef(let-void-2.18509) inst-val2.18512) (with-continuation-mark parameterization-key (let ([AppRand5438 (continuation-mark-set-first #f parameterization-key)][if33542 (eq? CellRef(let-void-0.18507) 'relative)]) (let ([AppRand5438 AppRand5438][AppRand5439 (if if33542 (current-directory) (path->complete-path CellRef(let-void-0.18507)))]) (extend-parameterization AppRand5438 current-load-relative-directory AppRand5439))) (let ([AppRator172 (LinkletVar(current-load:#<parameter-procedure>))]) (AppRator172 letone18506 #f)))))))
  in function: (lambda (lam.val.18505) (let ([if33540 (path? lam.val.18505)]) (if if33540 #f (let ([if33538 (string? lam.val.18505)]) (let ([if33539 (if if33538 (let ([if33537 (relative-path? lam.val.18505)]) (if if33537 #t (absolute-path? lam.val.18505))) #f)]) (if if33539 #f (raise-argument-error 'load path-string? lam.val.18505))))) (let ([if33541 (string? lam.val.18505)]) (let ([letone18506 (if if33541 (string->path lam.val.18505) lam.val.18505)]) (let ([let-void-0.18507 'uninitialized][let-void-1.18508 'uninitialized][let-void-2.18509 'uninitialized][(inst-val0.18510 inst-val1.18511 inst-val2.18512) (split-path letone18506)]) (set! CellRef(let-void-0.18507) inst-val0.18510) (set! CellRef(let-void-1.18508) inst-val1.18511) (set! CellRef(let-void-2.18509) inst-val2.18512) (with-continuation-mark parameterization-key (let ([AppRand5438 (continuation-mark-set-first #f parameterization-key)][if33542 (eq? CellRef(let-void-0.18507) 'relative)]) (let ([AppRand5438 AppRand5438][AppRand5439 (if if33542 (current-directory) (path->complete-path CellRef(let-void-0.18507)))]) (extend-parameterization AppRand5438 current-load-relative-directory AppRand5439))) (let ([AppRator172 (LinkletVar(current-load:#<parameter-procedure>))]) (AppRator172 letone18506 #f))))))))
RPython traceback:
  File "pycket_entry_point.c", line 47, in entry_point
  File "pycket_entry_point.c", line 1204, in actual_entry
  File "pycket_racket_entry.c", line 2818, in racket_entry
  File "pycket_base.c", line 545, in W_Object_call_interpret
  File "pycket_interpreter.c", line 22980, in interpret_one
  File "rpython_jit_metainterp.c", line 14972, in handle_jitexception_6
  File "pycket_interpreter_1.c", line 4736, in portal_6
  File "pycket_interpreter.c", line 30858, in LetCont_plug_reduce
Fatal RPython error: SchemeException
```

**Comment by cderici** [comment]
I bet this is something about the prompts again:

$ p -I racket/kernel/init
Welcome to Pycket v7.1.0.6
\> (map)
exn:fail : map expected at least two argument, got 0
\>

**Comment by samth** [comment]
Fixed by d53b73b7c2d03f1
